pallets/committee/src/lib.rs

```diff
@@ -380,48 +417,23 @@ pub mod pallet {
 
             Ok(().into())
         }
-    }
 
-    /// Initialize council members. Can only be done once.
-    /// Constituent members must be initialized later by voting by the council
-    impl<T: Config> InitializeMembers<AccountIdFor<T>> for Pallet<T> {
-        fn initialize_members(members: &[AccountIdFor<T>]) {
-            if !members.is_empty() {
-                assert_eq!(
-                    Members::<T>::iter().count(),
-                    0,
-                    "Members are already initialized!"
-                );
-                for member in members {
-                    Members::<T>::insert(member, MemberType::Council);
-                }
-            }
-        }
-    }
+        #[pallet::weight(10_000)] // TODO: Set weights
+        /// Add new constituent to the committee
+        ///
+        /// NOTE:
+        ///
+        /// This call can only be called after the approval of the committee
+        pub fn add_constituent(
+            origin: OriginFor<T>,
+            constituent: AccountIdFor<T>,
+        ) -> DispatchResultWithPostInfo {
+            T::ApprovedByCommitteeOrigin::ensure_origin(origin)?;
 
-    /// Used to add and remove constituent members.
-    /// Council members must be added by first adding them as constituents.
-    /// The existing council can then vote to add them as concil members
-    impl<T: Config> ChangeMembers<AccountIdFor<T>> for Pallet<T> {
-        fn change_members_sorted(
-            incoming: &[AccountIdFor<T>],
-            outgoing: &[AccountIdFor<T>],
-            _sorted_new: &[AccountIdFor<T>],
-        ) {
-            // Remove outgoing members from any currently active votes
-            for proposal_hash in ActiveProposals::<T>::get() {
-                Votes::<T>::mutate(&proposal_hash, |votes| {
-                    if let Some(votes) = votes {
-                        votes.remove_voters(outgoing); // mutates votes in place
-                    }
-                });
-            }
-            for leaving_member in outgoing {
-                Members::<T>::remove(leaving_member)
-            }
-            for member in incoming {
-                Members::<T>::insert(member, MemberType::Constituent);
-            }
+            Members::<T>::insert(constituent.clone(), MemberType::Constituent);
```

Comment:
We probably should we consider treating a `constituent` that is already a member, either council or constituent an error? Specifically the former case will result in a council member being stripped of their council privileges. This way it would be possible to demote a council member to constituent.